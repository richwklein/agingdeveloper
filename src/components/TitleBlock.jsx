import React from "react";
import {Box, Typography} from "@mui/material";
import PropTypes from "prop-types";

const TitleBlock = ({title, subtitle}) => {
  return (
    <Box component="header" sx={{mt: 1}}>
      <Typography variant="h3">{title}</Typography>
      {subtitle &&
        <Typography variant="subtitle" component={Box}>
          {subtitle}
        </Typography>
      }
    </Box>
  );
};
TitleBlock.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
};

export default TitleBlock;
