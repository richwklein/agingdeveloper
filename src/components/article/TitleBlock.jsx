import React from "react";
import {Box, Typography} from "@mui/material";
import PropTypes from "prop-types";

const TitleBlock = ({title, description}) => {
  return (
    <Box
      sx={{
        marginTop: 1,
      }}>
      <Typography variant="h2">{title}</Typography>
      <Typography variant="subtitle" component={Box}>
        {description}</Typography>
    </Box>
  );
};
TitleBlock.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
};

export default TitleBlock;
