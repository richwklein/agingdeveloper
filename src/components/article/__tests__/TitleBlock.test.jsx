import React from "react";
import {cleanup, render} from "@testing-library/react";
import TitleBlock from "../TitleBlock";

describe("TitleBlock", () => {
  afterEach(cleanup);

  const renderComponent = () => render(
      <TitleBlock
        title= "Tesing Gatsby with Jest"
        description="You can test with Jest with the best of them." />,
  );

  test("that it renders correctly.", () => {
    const {container} = renderComponent();
    expect(container.firstChild).toMatchSnapshot();
  });
});
