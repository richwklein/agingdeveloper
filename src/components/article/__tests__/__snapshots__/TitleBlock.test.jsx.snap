// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleBlock that it renders correctly. 1`] = `
<div
  class="MuiBox-root css-164r41r"
>
  <h2
    class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
  >
    Tesing Gatsby with Jest
  </h2>
  <div
    class="MuiTypography-root MuiTypography-subtitle MuiBox-root css-37hqdj-MuiTypography-root"
  >
    You can test with Jest with the best of them.
  </div>
</div>
`;
